src/commands.ts

```diff
@@ -477,6 +478,61 @@ function insertFunctionComment(workspaceContext: WorkspaceContext) {
     workspaceContext.commentCompletionProvider.insert(activeEditor, line);
 }
 
+function expandMacro(workspaceContext: WorkspaceContext) {
+    const activeEditor = vscode.window.activeTextEditor;
+    const document = activeEditor?.document;
+    const selection = activeEditor?.selection;
+    if (!document || !selection) {
+        return;
+    }
+    return workspaceContext.languageClientManager.useLanguageClient(async (client, token) => {
+        const params: MacroExpansionParams = {
+            textDocument: client.code2ProtocolConverter.asTextDocumentIdentifier(document),
+            range: client.code2ProtocolConverter.asRange(selection),
+        };
+        const expansions = await client.sendRequest(macroExpansionRequest, params, token);
+        if (expansions && expansions.length >= 1) {
+            // Present the macro expansion using a custom in-memory URI scheme
+            // TODO: This will currently reregister the provider for this scheme
+            // every time a new macro expansion is requested. It looks like the
+            // old expansions will be closed correctly, but perhaps there's
+            // a more elegant solution to this?
+            const scheme = "swift-macro-expansion";
+            const provider = vscode.workspace.registerTextDocumentContentProvider(scheme, {
```

Comment:
We will need to manage the text document content providers slightly better. Currently you just keep creating them, without cleaning up the unused ones. Adding them to the workspaceContext subscriptions means they get removed when the workspace is closed, but not before then.